centaur/src/main/scala/centaur/test/Test.scala

```diff
@@ -171,10 +171,50 @@ object Operations {
   }
 
   /**
-    * Polls until a specific status is reached. If a terminal status which wasn't expected is returned, the polling
-    * stops with a failure.
+    * Polls until a valid status is reached.
+    * If an unexpected terminal status is returned, the polling stops with a failure.
+    * If none of the expected statuses are returned in time, the polling also stops with a failure.
     */
-  def pollUntilStatus(workflow: SubmittedWorkflow, testDefinition: Workflow, expectedStatus: WorkflowStatus): Test[SubmittedWorkflow] = {
+  def expectSomeProgress(workflow: SubmittedWorkflow,
+                         testDefinition: Workflow,
+                         expectedStatuses: Set[WorkflowStatus],
+                         timeout: FiniteDuration): Test[SubmittedWorkflow] = {
+    new Test[SubmittedWorkflow] {
+      def status(remainingTimeout: FiniteDuration): IO[SubmittedWorkflow] = {
+        for {
+          workflowStatus <- CentaurCromwellClient.status(workflow)
+          mappedStatus <- workflowStatus match {
```

Comment:
Any reason you don't do `mappedStatus = workflowStatus` and get rid of all the non-IO `IO`'s in the `match`?

Reply:
Is it because the `IO.sleep` is captured in the IO and deferred?